src/msg/xio/XioPool.h

```diff
@@ -194,16 +200,33 @@ extern XioPoolStats xp_stats;
 static inline int xpool_alloc(struct xio_mempool *pool, uint64_t size,
 			      struct xio_mempool_obj* mp)
 {
+  // try to allocate from the xio pool
+  int r = xio_mempool_alloc(pool, size, mp);
+  if (r == 0) {
```

Comment:
Hi,

The current allocator strategy uses bounded pools, so in general, this
deals with allocation about the current Accelio pool capacity, for whatever
reason.

Since this interface does not consume registered memory, it's not critical
that Accelio serve the allocation, in the uncommon case.

Matt

----- "Haomai Wang" <notifications@github.com> wrote:

> In src/msg/xio/XioPool.h : > @@ -194,16 +200,33 @@ extern XioPoolStats
> xp_stats;
> >  static inline int xpool_alloc(struct xio_mempool *pool, uint64_t
> size,
> >  			      struct xio_mempool_obj* mp)
> >  {
> > +  // try to allocate from the xio pool
> > +  int r = xio_mempool_alloc(pool, size, mp);
> > +  if (r == 0) {
> 
> Maybe it's a naive question, why we will failed to alloc from xio
> mempool? AFAR, accelio could extend new memory block and return new
> alloced memory inline?
> 
> —
> Reply to this email directly or view it on GitHub .

-- 
Matt Benjamin
CohortFS, LLC.
315 West Huron Street, Suite 140A
Ann Arbor, Michigan 48103

http://cohortfs.com

tel.  734-761-4689 
fax.  734-769-8938 
cel.  734-216-5309